### PR TITLE
[do not merge][Tests-Only] add tests for users with zero quota.

### LINKED ISSUE #38592: Test coverage for user with zero quota
Add API test coverage for the problem fixed in PR #38591 

A user that has zero quota (or not sufficient quota) tries to move a file from a received share into their own folder. The move should not be allowed.

Think of other scenarios where a user can have received access to a file and try to move it into their own folder, and test those.

Check existing test scenarios about quota and make sure they are covered well (e.g. uploading a file when the user does not have enough quota)

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Needs https://github.com/owncloud/core/pull/38591 to be merged first
add tests for users with zero quota.

## Related Issue

- Tests for https://github.com/owncloud/core/pull/38591
- Fixes https://github.com/owncloud/core/issues/38592

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiMain/quota.feature [hunk 1]
```diff
@@ -100,3 +100,93 @@ Feature: quota
     When user "Brian" overwrites from file "filesForUpload/textfile.txt" to file "/testquota.txt" with all mechanisms using the WebDAV API
     Then the HTTP status code of all upload responses should be "507"
     And the content of file "/testquota.txt" for user "Alice" should be "test"
+
+
+  Scenario: User with zero quota cannot upload a file
+    Given user "Brian" has been created with default attributes and without skeleton files
+    And the quota of user "Alice" has been set to "0 B"
+    When user "Alice" uploads file with content "uploaded content" to "testquota.txt" using the WebDAV API
+    Then the HTTP status code should be "507"
```
[the remark this answers is not in the document]
Comment (karakayasemi) [reply tests/acceptance/features/apiMain/quota.feature:109]: Yes, it is correct. Adjusted frontend in here with the same code https://github.com/owncloud/core/pull/38591/files#diff-dedceb236d4709ee562b67512f8ad10bce9dc7a494e9b1cfdabe8ea8ad33ca25R2262

File: tests/acceptance/features/apiMain/quota.feature [hunk 2]
```diff
@@ -100,3 +100,93 @@ Feature: quota
...
+    And the content of file "/shareFolder/newTextFile.txt" for user "Alice" should be "uploaded content"
+
+
+  Scenario: User should retain their old files even if their quota is set to 0
+    And user "Alice" has uploaded file with content "test" to "/testquota.txt"
+    And the quota of user "Alice" has been set to "0 B"
+    And the content of file "/testquota.txt" for user "Alice" should be "test"
+
+
+  Scenario: User should be able to restore their deleted file when their quota is set to zero
```
[the remark this answers is not in the document]
Comment (VicDeo) [reply tests/acceptance/features/apiMain/quota.feature:186]: @phil-davis 
>Quotas Are Neither Hard Nor Soft
Quotas are not strictly hard quotas, but nor are they soft quotas either. They’re more hard quotas with the exception of always being able to restore deleted files.

https://doc.owncloud.com/server/user_manual/files/webgui/quota.html#quotas-are-neither-hard-nor-soft

[the remark this answers is not in the document]
Comment (phil-davis) [reply tests/acceptance/features/apiMain/quota.feature:186]: Thanks - good to see that the behaviour is documented.

File: tests/acceptance/features/apiMain/quota.feature [hunk 3]
```diff
@@ -100,3 +100,93 @@ Feature: quota
...
+    And user "Alice" has shared file "/shareFolder" with user "Brian"
+    And user "Brian" has accepted share "/shareFolder" offered by user "Alice"
+    When user "Brian" uploads file with content "uploaded content" to "/Shares/shareFolder/newTextFile.txt" using the WebDAV API
+    Then the HTTP status code should be "201"
+    And the content of file "/shareFolder/newTextFile.txt" for user "Alice" should be "uploaded content"
+
+
+  Scenario: User should retain their old files even if their quota is set to 0
+    And user "Alice" has uploaded file with content "test" to "/testquota.txt"
+    And the quota of user "Alice" has been set to "0 B"
```
[the remark this answers is not in the document]
Comment (phil-davis) [reply tests/acceptance/features/apiMain/quota.feature:180]: The `When` step is:
```
When the administrator changes the quota of user "Alice" to "0 B" using the provisioning API
```

But actually, it seems a bit strange to specify "the provisioning API" or "the occ command" here. We are not specifically testing either of those particular ways to set the quota. We just want to test that "when the quota has been reset after a file already exists, then the file still exists (without anything else happening)". So the current steps look OK to me - the "When" step is "When nothing else happens" :)

**Review by phil-davis — COMMENTED** [review]
@dpakach the tests LGTM now. You could cherry-pick this into the original PR and that way the tests will get merged along with the code changes.

**Comment by phil-davis** [comment]
This has been cherry-picked into #38591

**Pull request closed without merging**